portfolio/src/main/webapp/script.js

```diff
@@ -12,40 +12,55 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+let index = 0;
+let timer;
+
 /**
- * Adds a random quote to the page.
+ * Set slide index to next and call show.
  */
+function nextSlide() {
+    index = (index + 1) % 4;
```

Comment:
The remainder operator is so that the slideshow will return to slide 0 after reaching the last slide. Should I make this more in the comments, or should I go about this in a different way? I could use an if statement to check if its on the last slide, or I could even remove this function and modify the index at the end of showSlide.

Reply:
Ah, clever. You'll want to utilize `slides.length` as part of this chain of operations, instead of hardcoding the number.